plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/FailureAwareThriftMetastoreClient.java

```diff
@@ -90,17 +90,17 @@ public List<String> getAllTables(String databaseName)
     }
 
     @Override
-    public List<String> getAllViews(String databaseName)
+    public List<String> getTableNamesByFilter(String databaseName, String filter)
```

Comment:
This commit is not "Move alternate calls logic to ThriftHiveMetastore", but a massive refactor of the APIs for the ThriftMetastoreClient.  If we want to rename methods, those should be move to a separate commit.

Reply:
> This commit is not "Move alternate calls logic to ThriftHiveMetastore", but a massive refactor of the APIs for the ThriftMetastoreClient

that's mostly a revert of some previous refactor that turned out to be harmful
so the current "massive refactor" doesn't introduce any new quality that we haven't had before

Reply:
I couldn't do a simple revert, as these classes has been changed along the way. But yes, this is just a revert, no additional logic is implemented here.
IMHO following the single responsibility principle, the *Client class should just wrap the underlying thrift code, without any logic.
Another thing is that if, for any reason, the change that's being reverted here does add any value, we can simply put a config option that would decide whether the alternate calls should reuse the client.

Reply:
> put a config option that would decide whether the alternate calls should reuse the client.

there is no benefit to that, only downsides, so let's not do that